src/storage/src/mem_table.rs

```diff
@@ -151,21 +150,36 @@ impl MemTable {
             }
             Entry::Occupied(mut e) => {
                 let old_op = e.get_mut();
-                self.kv_size.sub_val(old_op);
+                let old_op_size = old_op.estimated_size();
```

Comment:
`old_op` size will be `sub_size` anyway in either case. May do remain the same logic as original. When error happens, the correctness of this size can be ignored.